src/html_files/cpu_utilization.ts

```diff
@@ -1,6 +1,50 @@
 let got_cpu_util_data = false;
 let util_cpu_list: Map<string, CPUList> = new Map<string, CPUList>();
 
+let cpu_utilization_rules = {
+    data_type: "cpu_utilization",
+    pretty_name: "CPU Utilization",
+    rules: [
+        {
+            name: "User",
+            single_run_rule: function* (opts): Generator<Finding, void, any> {
```

Comment:
Just thinking a bit more on this, `all_run_rule` and `single_run_rule` are not much different.  If I'm looking at CPU utilization, I'd be just as interested knowing all have low utilization in addition to just the base run.  We can eliminate the "single run" case and just apply the rules that look at all runs independently.

Reply:
The aim is to logically separate 3 use cases. Consider 3 runs A, B and C. all_run_rule will allow us to compare the data across all runs. per_run_rule allows us to compare A B and A C. single_run_rule will allow us to write rules for each run. This will apply when comparison is not possible when a report is generated for a single run in addition to applying it for multiple runs. Ex: A config is KConfig is known to cause performance regressions. We want that flagged in both cases. The split is a logical separation helper for implementers.

Reply:
Additionally, each single_run_rule will need an if check to make sure the other comparison run exists. If we have 3 separate rule calls it allows basic assumptions.